packages/node_modules/@node-red/nodes/core/sequence/17-split.js

```diff
@@ -603,6 +603,15 @@ module.exports = function(RED) {
                     return;
                 }
 
+                if (node.mode === 'custom' && msg.hasOwnProperty('parts')) {
+                    if (msg.parts.hasOwnProperty('parts')) {
+                        msg.parts = { parts: msg.parts.parts };
+                    }
```

Comment:
I've stared at this code for a while and I *think* this doesn't actually result in a different object - I think the structure is identical.

Reply:
It should be cleaning  msg.parts to just leave msg.parts.parts. (IE unravelling a split of splits) - BUT removing all existing (top level) parts so it doesn't auto join at that level. In case for some reason the user wants to auto join the "outer" split.